serdes/serde-common/src/main/java/io/apicurio/registry/serdes/SchemaMapper.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.serdes;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+import io.apicurio.registry.types.ArtifactType;
+import io.apicurio.registry.utils.IoUtil;
+
+/**
+ * @author Fabian Martinez
+ */
+public interface SchemaMapper<S, D> {
```

Comment:
Is this ever used for non-Avro types?

Reply:
changed to SchemaParser, now it makes more sense and it's really necessary for the SchemaResolver